### DISCUSSION TITLE
minimal config for hyprland with nvidia on manjaro

### DISCUSSION THREAD

**Discussion by ExpandingMan** [hyprwm/Hyprland#3563] [Q&A]
Wayland now "just works" with all my nvidia cards on `qtile`, so I thought it was about time to see if I could get this running since  my transition to wayland seems finally (finally) in sight.  I use Manjaro on pretty much everything, so my goal here is not just to get `hyprland` working but also to figure out the bare minimal shenanigans required to do that.

As such, the first thing I tried was to just run it out of the box (amazingly, this works for `qtile` now).  That doesn't work.  No surprise here as the docs tell me this is going to happen.

Next, I add these lines from the docs to the hyprland config.
```
env = LIBVA_DRIVER_NAME,nvidia
env = XDG_SESSION_TYPE,wayland
env = GBM_BACKEND,nvidia-drm
env = __GLX_VENDOR_LIBRARY_NAME,nvidia
env = WLR_NO_HARDWARE_CURSORS,1
```

Still doesn't work, logs are telling me `m_sWLRBackend was NULL!` which seems to be the critical error at which it gives up (this is the same thing that happened before putting in these config entries btw).

Next thing I try is to create `/etc/modprobe.d/nvidia.conf` and write `options nvidia-drm modeset=1` (again, the reason I didn't do this in the first place is because I'm trying to figure out the minimal config).  This seems to break `lightdm`.  Ignoring that, when I run `Hyprland`, the `m_sWLRBackend` error now disappears, but I get `Could not connect to socket /run/seatd.sock: No such file or directory`.  The reference to "seat" makes me suspect this is because I broke `lightdm`, but the messages don't make this seem like a critical error.  Below it says `Successfully loaded libseat session` and `Found 2 GPUs`, but after that nothing happens.

Now I have reached a point where I am slightly less comfortable taking the suggested steps (because I don't understand very well how `grub` and `mkinitcpio` work.  The next thing I would try would be to edit `/etc/default/grub`, and generate the new grub config.  The docs suggest I also modify `/etc/mkinitcpio.conf` and generate it.  Not knowing how either of these work I don't know if it makes sense to attempt changing one at a time.

Before I do this, is there any advice about what's going on at this point?  Is what I reported above consistent with expectations?  Does anyone else have experience doing a minimal setup on Manjaro for nvidia?  Again, at this phase I'm at least as worried about trying to sort out a standard way of doing this where I can have confidence everything makes sense as I am about actually getting it working (i.e. if I get it working but did too many shenanigans and am not confident I can reproduce them, I'll never use it).

Thanks!

**Comment by Aqa-Ib** [comment]
You might try with the Nouveau driver. For me (i dont game), it is the best. Everything just works out of the box, without tweaking anything at all.
